orbit/models/ktrx.py

```diff
@@ -453,38 +453,37 @@ def _set_coefficients_kernel_matrix(self, df):
-
         if self._num_of_positive_regressors > 0:
-            # store local value for the range on the left side since last knot
-            for idx in range(len(self._knots_idx_coef)):
-                if idx < len(self._knots_idx_coef) - 1:
-                    str_idx = self._knots_idx_coef[idx]
-                    end_idx = self._knots_idx_coef[idx + 1]
-                else:
-                    str_idx = self._knots_idx_coef[idx]
-                    end_idx = self.num_of_observations
-
-                local_val[:, idx] = np.mean(np.fabs(self._positive_regressor_matrix[str_idx:end_idx]), axis=0)
-
-            # adjust knot scale with the multiplier derive by the average value and shift by 0.001 to avoid zeros in
-            # scale parameters
-            local_min = np.amin(local_val, axis=-1, keepdims=True)
-            local_max = np.amax(local_val, axis=-1, keepdims=True)
-            multiplier = (
-                    (local_val - local_min) / (local_max - local_min) *
-                    (DEFAULT_UPPER_BOUND_SCALE_MULTIPLIER - DEFAULT_LOWER_BOUND_SCALE_MULTIPLIER) +
-                    DEFAULT_LOWER_BOUND_SCALE_MULTIPLIER
-            )
-            # replace entire row of nan (when local_max is equal to local_min) with upper bound
-            multiplier[np.isnan(multiplier).all(axis=-1)] = DEFAULT_UPPER_BOUND_SCALE_MULTIPLIER
+            # calculate average local absolute volume for each segment
+            local_val = np.ones((self._num_of_positive_regressors, self._num_knots_coefficients))
             if self.flat_multiplier:
-                multiplier = np.ones(multiplier.shape)
+                multiplier = np.ones(local_val.shape)
+            else:
+                multiplier = np.ones(local_val.shape)
+                # store local value for the range on the left side since last knot
+                for idx in range(len(self._knots_idx_coef)):
+                    if idx < len(self._knots_idx_coef) - 1:
+                        str_idx = self._knots_idx_coef[idx]
+                        end_idx = self._knots_idx_coef[idx + 1]
+                    else:
+                        str_idx = self._knots_idx_coef[idx]
+                        end_idx = self.num_of_observations
+
+                    local_val[:, idx] = np.mean(np.fabs(self._positive_regressor_matrix[str_idx:end_idx]), axis=0)
+
+                # adjust knot scale with the multiplier derive by the average value and shift by 0.001 to avoid zeros in
+                # scale parameters
+                global_med = np.expand_dims(np.mean(np.fabs(self._positive_regressor_matrix), axis=0), -1)
+                test_flag = local_val < 0.01 * global_med
+
+                multiplier[test_flag] = DEFAULT_LOWER_BOUND_SCALE_MULTIPLIER
+                # replace entire row of nan (when 0.1 * global_med is equal to global_min) with upper bound
+                multiplier[np.isnan(multiplier).all(axis=-1)] = 1.0
```

Comment:
here it is intended to set 1.0 hard-coded or we can use `DEFAULT_LOWER_BOUND_SCALE_MULTIPLIER`?